stripe/src/test/java/com/stripe/android/StripeKtxTest.kt

```diff
@@ -0,0 +1,561 @@
+
+            assertFailsWith<InvalidRequestException> {
+                stripe.confirmPaymentIntent(mock())
+            }
+        }
+
+    private inline fun <reified APIObject : StripeModel, reified CreateAPIParam : Any, reified RepositoryParam : Any>
+    `Given repository returns non-empty value when calling createAPI then returns correct result`(
+        crossinline repositoryInvocationBlock: suspend (RepositoryParam, ApiRequest.Options) -> APIObject?,
+        crossinline createAPIInvocationBlock: suspend (CreateAPIParam, String?, String?) -> APIObject
+    ) = testDispatcher.runBlockingTest {
+        val expectedApiObj = mock<APIObject>()
+
+        whenever(
+            repositoryInvocationBlock(any(), any())
+        ).thenReturn(expectedApiObj)
+
+        val actualObj = createAPIInvocationBlock(
+            mock(),
+            TEST_IDEMPOTENCY_KEY,
+            TEST_STRIPE_ACCOUNT_ID
+        )
+
+        assertSame(expectedApiObj, actualObj)
+    }
+
+    private inline fun <reified CreateAPIParam : Any, reified RepositoryParam : Any>
+    `Given repository throws exception when calling createAPI then throws same exception`(
+        crossinline repositoryInvocationBlock: suspend (RepositoryParam, ApiRequest.Options) -> StripeModel?,
+        crossinline createAPIInvocationBlock: suspend (CreateAPIParam, String?, String?) -> StripeModel
+    ): Unit = testDispatcher.runBlockingTest {
+        whenever(
+            repositoryInvocationBlock(any(), any())
+        ).thenThrow(mock<AuthenticationException>())
+
+        assertFailsWith<AuthenticationException> {
+            createAPIInvocationBlock(
+                mock(),
+                TEST_IDEMPOTENCY_KEY,
+                TEST_STRIPE_ACCOUNT_ID
+            )
+        }
+    }
+
+    private inline fun <reified CreateAPIParam : Any, reified RepositoryParam : Any>
+    `Given repository returns empty value when calling createAPI then thorws APIException`(
+        crossinline repositoryInvocationBlock: suspend (RepositoryParam, ApiRequest.Options) -> StripeModel?,
+        crossinline createAPIInvocationBlock: suspend (CreateAPIParam, String?, String?) -> StripeModel
+    ): Unit = testDispatcher.runBlockingTest {
+        whenever(
+            repositoryInvocationBlock(any(), any())
+        ).thenReturn(null)
+
+        assertFailsWith<InvalidRequestException> {
+            createAPIInvocationBlock(
+                mock(),
+                TEST_IDEMPOTENCY_KEY,
+                TEST_STRIPE_ACCOUNT_ID
+            )
+        }
+    }
+
+    private inline fun <reified APIObject : StripeModel, reified RepositoryParam : Any>
+    `Given repository returns non-empty value when calling createAPI with String param then returns correct result`(
+        crossinline repositoryInvocationBlock: suspend (RepositoryParam, ApiRequest.Options) -> APIObject?,
+        crossinline createAPIInvocationBlock: suspend (String, String?, String?) -> APIObject
```

Comment:
Nit: can we just name it `repositoryBlock`?

Also `Api` instead of `API` (e.g. `createApiBlock`)?